File: docs/guides/csharp/src/searchWithGAToken.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SearchWithGAToken
{
  private static string GetGoogleAnalyticsUserIdFromBrowserCookie(string cookieName)
  {
    return ""; // Implement your logic here
  }

  async Task Main(string[] args)
  {
    try
    {
      var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));
      var userToken = GetGoogleAnalyticsUserIdFromBrowserCookie("_ga");
      var searchParams = new SearchParams(
        new SearchParamsObject { Query = "<YOUR_SEARCH_QUERY>", UserToken = userToken }
      );

      await client.SearchSingleIndexAsync<Hit>("<YOUR_INDEX_NAME>", searchParams);

      string? loggedInUser = null;
      searchParams.AsSearchParamsObject().UserToken = loggedInUser ?? userToken;

      await client.SearchSingleIndexAsync<Hit>("<YOUR_INDEX_NAME>", searchParams);
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}

File: docs/guides/dart/lib/searchWithGAToken.dart
import 'package:algolia_client_search/algolia_client_search.dart';

String getGoogleAnalyticsUserIdFromBrowserCookie(String cookieName) {
  // This function is a placeholder for a real implementation
  return "";
}

void searchWithGAToken() async {
  final client =
      SearchClient(appId: 'ALGOLIA_APPLICATION_ID', apiKey: 'ALGOLIA_API_KEY');

  final userToken = getGoogleAnalyticsUserIdFromBrowserCookie("_ga");
  var searchParams =
      SearchParamsObject(query: "<YOUR_SEARCH_QUERY>", userToken: userToken);

  await client.searchSingleIndex(
    indexName: "<YOUR_INDEX_NAME>",
    searchParams: searchParams,
  );

  String? loggedInUser;
  searchParams = SearchParamsObject(
      query: "<YOUR_SEARCH_QUERY>", userToken: loggedInUser ?? userToken);

  await client.searchSingleIndex(
    indexName: "<YOUR_INDEX_NAME>",
    searchParams: searchParams,
  );
}
